istioctl/cmd/proxyconfig.go

```diff
@@ -190,6 +303,94 @@ func proxyConfig() *cobra.Command {
 	listenerConfigCmd.PersistentFlags().StringVar(&listenerType, "type", "", "Filter listeners by type field")
 	listenerConfigCmd.PersistentFlags().IntVar(&port, "port", 0, "Filter listeners by Port field")
 
+	loggingCmd := &cobra.Command{
+		Use:   "logging <pod-name[.namespace]>",
+		Short: "(experimental) Retrieves logging levels of the Envoy in the specified pod",
+		Long:  "(experimental) Retrieve information about logging levels of the Envoy instance in the specified pod, and update optionally",
+		Example: `  # Retrieve information about logging levels for a given pod from Envoy.
+  istioctl proxy-config logging <pod-name[.namespace]>
+
+  # Update levels of the specified loggers and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> --log_level=all:debug
+
+  # Reset levels of all the loggers to default value (warning) and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> -r
+`,
+		Aliases: []string{"o"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("logging requires pod name")
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			destLoggerLevels := map[string]Level{}
+			if reset {
+				// reset logging level to `defaultOutputLevel`, and ignore the `--log_level` option
+				destLoggerLevels[DefaultLoggerName] = defaultOutputLevel
+			} else if logLevelString != "" {
```

Comment:
I think we need some validation logic here. I tried `--log_level http=info` and nothing happened -- there was no error and no effect.

Reply:
+1 for validation. Looks like envoy returns 404 for invalid logger/level combinations, but it doesn't provide details on what exactly is wrong. 

Instead of hard-coding loggers in istioctl, you could validate against the list of available loggers discovered by querying the target envoy.

Reply:
I adjusted the logic and reported an error when specified `logger:level` was wrong.